Image not building due to old rust-lstm version tag

Failure on building rust-lstm during ngen image build. Dockerfile pulls rust-lstm v0.1.0, but pulls aaraney/bmi-rs which is newer and has[ removed the default implementation](https://github.com/aaraney/bmi-rs/commit/93dbccb27dea2d05f07b38d30f10f2a708c96f77) of `get_var_nbytes`, which the rust-lstm tagged version still relies on. The `get_var_nbytes` issue is fixed in a [later rust-lstm PR](https://github.com/CIROH-UA/rust-lstm/pull/2) but hasn't been tagged in a version yet.

## Current behavior

Image build fails with error below

## Expected behavior

Image builds with fixed rust-lstm (or older bmi-rs)

## Steps to replicate behavior (include URLs)

1. Pull latest NGIAB-CloudInfra
2. Build latest image

## Screenshots

```
...
   Compiling serde_yaml v0.9.34+deprecated                                                                                                                                                                                    
   Compiling burn-import v0.18.0                                                                                                                                                                                              
   Compiling bmi-rs v0.0.1-alpha.0 (/build/bmi-rs/bmi-rs)                                                                                                                                                                     
   Compiling rust-lstm-1025 v0.1.0 (/build/rust-lstm-1025)                                                                                                                                                                    
error[E0046]: not all trait items implemented, missing: `get_var_nbytes`                                                                                                                                                      
   --> src/lib.rs:403:1                                                                                                                                                                                                       
    |                                                                                                                                                                                                                         
403 | impl<B: Backend> Bmi for LstmBmi<B> {                                                                                                                                                                                   
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^ missing `get_var_nbytes` in implementation                                                                                                                                          
    |                                                                                                                                                                                                                         
    = help: implement the missing item: `fn get_var_nbytes(&self, _: &str) -> Result<u32, Box<(dyn StdError + 'static)>> { todo!() }`                                                                                         
                                                                                                                                                                                                                              
warning: unused variable: `output_size`                                                                                                                                                                                       
   --> src/nextgen_lstm.rs:133:13                                                                                                                                                                                             
    |                                                                                                                                                                                                                         
133 |         let output_size = weights["output_size"].as_u64().unwrap() as usize;                                                                                                                                            
    |             ^^^^^^^^^^^ help: if this is intentional, prefix it with an underscore: `_output_size`                                                                                                                      
    |                                                                                                                                                                                                                         
    = note: `#[warn(unused_variables)]` (part of `#[warn(unused)]`) on by default                                                                                                                                             
                                                                                                                                                                                                                              
For more information about this error, try `rustc --explain E0046`.                                                                                                                                                           
warning: `rust-lstm-1025` (lib) generated 1 warning                                                                                                                                                                           
error: could not compile `rust-lstm-1025` (lib) due to 1 previous error; 1 warning emitted
[10/12] STEP 1/5: FROM 16b2edbe31e8b0de6f8ed2741d36373a4ab031a5c6abf56d787ef26cc320974e AS lstm_weights
[10/12] STEP 2/5: RUN git clone --depth=1 --branch example_weights https://github.com/ciroh-ua/lstm.git /lstm_weights
Error: building at STEP "RUN cargo build --release": while running runtime: exit status 101```
```

--- a/docker/Dockerfile
+++ b/docker/Dockerfile
@@ -31,7 +31,7 @@ RUN dnf install -y epel-release && \
     python3 python3-devel python3-pip \
     expat-devel flex bison udunits2-devel zlib-devel \
     wget mpich-devel hdf5-devel netcdf-devel \
-    netcdf-fortran-devel netcdf-cxx-devel lld
+    netcdf-fortran-devel netcdf-cxx-devel lld clang
 
 
 
@@ -181,8 +181,7 @@ RUN dnf install -y clang
 WORKDIR /build
 RUN curl https://sh.rustup.rs -sSf | bash -s -- -y
 RUN echo 'source $HOME/.cargo/env' >> $HOME/.bashrc
-RUN git clone --depth=1 https://github.com/aaraney/bmi-rs
-RUN git clone --depth=1 --branch v0.1.0 https://github.com/ciroh-ua/rust-lstm-1025
+RUN git clone --depth=1 --branch v0.1.2 https://github.com/ciroh-ua/rust-lstm-1025
 WORKDIR /build/rust-lstm-1025
 RUN cargo build --release
 
@@ -235,4 +234,4 @@ COPY --from=ngen_build /tmp/ngen_url /ngen/ngen_url
 
 RUN echo "export PS1='\u\[\033[01;32m\]@ngiab_dev\[\033[00m\]:\[\033[01;35m\]\W\[\033[00m\]\$ '" >> ~/.bashrc
 
-ENTRYPOINT ["./HelloNGEN.sh"]
+ENTRYPOINT ["/ngen/HelloNGEN.sh"]

